packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -299,95 +290,71 @@ export function getNetworkLogo(chainId: number) {
   }
 }
 
-export type SwitchChainProps = {
-  chainId: number
-  provider: ExtendedWeb3Provider
-  onSuccess?: () => void
-  onError?: (err?: Error) => void
-  onSwitchChainNotSupported?: (attemptedChainId: number) => void
-}
-
-const isSwitchChainSupported = (provider: ExtendedWeb3Provider) => {
-  const { provider: innerProvider } = provider
-  return innerProvider.isMetaMask || innerProvider.isImToken
+export function getSupportedNetworks(chainId = 0) {
+  return isNetwork(chainId).isTestnet
+    ? [ChainId.Goerli, ChainId.ArbitrumGoerli]
+    : [ChainId.Mainnet, ChainId.ArbitrumOne, ChainId.ArbitrumNova]
 }
 
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const noop = () => {}
-
-const onSwitchChainNotSupportedDefault = (attemptedChainId: number) => {
+const onSwitchChainNotSupported = (
+  attemptedChainId: number,
+  isAttemptingToPerformTx: boolean
```

Comment:
Love this! Would suggest a change to the name to make it clearer: `isSwitchingNetworkBeforeTx`